src/commands/appSettings/filterDownloadAppSettings.ts

```diff
@@ -11,42 +11,75 @@ import { localize } from "../../localize";
 
     for (const key of Object.keys(sourceSettings)) {
-        if (destinationSettings[key] === undefined) {
-            addedKeys.push(key);
-            destinationSettings[key] = sourceSettings[key];
-        } else if (destinationSettings[key] === sourceSettings[key]) {
-            matchingKeys.push(key);
-        } else if (sourceSettings[key]) { // ignore empty settings
-            if (!suppressPrompt) {
+        if (listOfSettingsToIgnore.includes(key)) {
+            if (!suppressPromptOfSettingsToIgnore) {
                 const yesToAll: vscode.MessageItem = { title: localize('yesToAll', 'Yes to all') };
                 const noToAll: vscode.MessageItem = { title: localize('noToAll', 'No to all') };
-                const message: string = localize('overwriteSetting', 'Setting "{0}" already exists in "{1}". Overwrite?', key, destinationName);
+                const message: string = localize('overwriteSetting', 'Setting "{0}" has been identified as unsafe. Download?', key);
                 const result: vscode.MessageItem = await context.ui.showWarningMessage(message, { modal: true }, DialogResponses.yes, yesToAll, DialogResponses.no, noToAll);
                 if (result === DialogResponses.yes) {
-                    overwriteSetting = true;
+                    overwriteSettingsToIgnore = true;
                 } else if (result === yesToAll) {
-                    overwriteSetting = true;
-                    suppressPrompt = true;
+                    overwriteSettingsToIgnore = true;
+                    suppressPromptOfSettingsToIgnore = true;
                 } else if (result === DialogResponses.no) {
-                    overwriteSetting = false;
+                    overwriteSettingsToIgnore = false;
                 } else if (result === noToAll) {
-                    overwriteSetting = false;
-                    suppressPrompt = true;
+                    overwriteSettingsToIgnore = false;
+                    suppressPromptOfSettingsToIgnore = true;
                 }
-            }
 
-            if (overwriteSetting) {
-                updatedKeys.push(key);
+                if (overwriteSettingsToIgnore) {
+                    updatedKeys.push(key);
+                    destinationSettings[key] = sourceSettings[key];
+                } else {
+                    securitySettingsIgnored.push(key);
+                    destinationSettings[key] = "";
+                }
+            }
+        }
+        else {
+            if (destinationSettings[key] === undefined) {
+                addedKeys.push(key);
                 destinationSettings[key] = sourceSettings[key];
-            } else {
-                userIgnoredKeys.push(key);
+            } else if (destinationSettings[key] === sourceSettings[key]) {
+                matchingKeys.push(key);
+            } else if (sourceSettings[key]) { // ignore empty settings
```

Comment:
I dont understand the comment `// ignore empty settings` because your condition `if(sourcesettingsp[key])` will only succeed IF it is not empty.

Reply:
That was a misplaced comment from the original flow so I will just removed it.